xdsl/builder.py

```diff
@@ -340,7 +340,7 @@ class ImplicitBuilder(contextlib.AbstractContextManager[tuple[BlockArgument, ...
     """
 
     _stack: ClassVar[_ImplicitBuilderStack] = _ImplicitBuilderStack()
```

Comment:
why is there still a stack? I expected an optional parent object to replace the current one with on exit

Reply:
We can improve it in a different PR